accounts/views.py

```diff
@@ -73,7 +76,31 @@ def get(self, request):
             'detail': serializer.errors
         }, status=status.HTTP_400_BAD_REQUEST)
             
-            
+
+# user view
+class UserDetailView(RetrieveAPIView):
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+
+    def get_object(self):
+        token = self.kwargs.get('token')
+        try:
+            user = User.objects.get(auth_token=token)
+            return user
+        except User.DoesNotExist:
+            raise NotFound({
+                'status': 'error',
+                'detail': 'User not found.'
+            })
+
+    def retrieve(self, request, *args, **kwargs):
+        user = self.get_object()
+        serializer = self.get_serializer(user)
+        return Response({
+            'status': 'success',
+            'detail': 'The user was found!',
+            'user': serializer.data
+        })
```

Comment:
A function that shows us the successful result using the user data included in the serializer.